src/openforms/submissions/templates/report/submission_report.html

```diff
@@ -27,7 +27,11 @@
         {% include 'includes/design-tokens.html' with skip_csp=True %}
     </head>
 
-    {% if config.organization_name %}
+    {% if theme.organization_name %}
+        {% blocktranslate with name=theme.organization_name asvar logo_alt trimmed %}
+            Logo {{ name }}
+        {% endblocktranslate %}
+    {% elif config.organization_name %}
```

Comment:
You can restructure some bits here with `{% firstof theme.organization_name config.organization_name as org_name %}` and then use the `org_name` variable.